### PR TITLE
HODS-433 | Implemented new mongo implementation

### PR DESCRIPTION
**Pull request by matthewharris1**

### CODE REVIEW

File: test/unit/uk/gov/hmrc/organisationsdetailsapi/services/CacheServiceSpec.scala [hunk 1]
```diff
@@ -54,25 +51,6 @@ class CacheServiceSpec extends AnyWordSpec with Matchers with MockitoSugar with
   }
 
   "cacheService.get" should {
-
-    "return the cached value for a given id and key" in new Setup {
```
Comment (parkeradam) [inline test/unit/uk/gov/hmrc/organisationsdetailsapi/services/CacheServiceSpec.scala:58]: Is there a reason these tests are deleted? They seem like reasonably valuable tests?

File: app/uk/gov/hmrc/organisationsdetailsapi/cache/ShortLivedCache.scala [hunk 2]
```diff
@@ -16,92 +16,82 @@
...
+
+import java.time.{LocalDateTime, ZoneOffset}
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ShortLivedCache @Inject()(
-                                 val cacheConfig: CacheConfiguration,
+class ShortLivedCache @Inject() (val cacheConfig: CacheRepositoryConfiguration,
```
Comment (ajchilts) [inline app/uk/gov/hmrc/organisationsdetailsapi/cache/ShortLivedCache.scala:37]: Why are we keeping this class when we have added the new cache repository?

**Review by ajchilts — CHANGES_REQUESTED** [review]

File: test/it/uk/gov/hmrc/organisationsdetailsapi/cache/ShortLivedCacheSpec.scala [hunk 3]
```diff
@@ -16,34 +16,32 @@
...
 import play.api.libs.json.{JsString, Json, OFormat}
+import uk.gov.hmrc.mongo.play.json.Codecs.toBson
 import uk.gov.hmrc.mongo.test.MongoSupport
 import uk.gov.hmrc.organisationsdetailsapi.cache.ShortLivedCache
 import utils.TestSupport
 
+import java.util.UUID
 import scala.concurrent.ExecutionContext
 
 class ShortLivedCacheSpec
```
Comment (ajchilts) [inline test/it/uk/gov/hmrc/organisationsdetailsapi/cache/ShortLivedCacheSpec.scala:33]: We have created a cache repository so we do not need this test class? Do we have one to cover cache repository?

**Review by ajchilts — CHANGES_REQUESTED** [review]

**Review by ajchilts — APPROVED** [review]

File: app/uk/gov/hmrc/organisationsdetailsapi/cache/CacheRepository.scala [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    .getOptional[Int](
+      "cache.ttlInSeconds"
+    )
+    .getOrElse(60 * 15)
+
+  lazy val collName = configuration
+    .getOptional[String](
+      "cache.collName"
+    )
+    .getOrElse("individuals-details-cache")
```
Comment (Luka-Abey) [inline app/uk/gov/hmrc/organisationsdetailsapi/cache/CacheRepository.scala:119]:
```suggestion
    .getOrElse("organisations-details-cache")
```

**Review by Luka-Abey — CHANGES_REQUESTED** [review]

**Review by parkeradam — APPROVED** [review]

**Review by Luka-Abey — APPROVED** [review]

**Pull request merged**